accepted/2021/statics-in-interfaces/README.md

```diff
@@ -6,8 +6,1267 @@
+    {
+    }
+
+    public struct Guid
+        : IComparableOperators<Guid, Guid>,
+          ISpanFormattable,
+          ISpanParseable<Guid>
+    {
+    }
+
+    public struct Half
+        : IBinaryFloatingPoint<Half>,
+          IFixedWidth<Half>
+    {
+    }
+
+    public struct Int16
+        : IBinaryInteger<short>,
+          IConvertible,
+          IFixedWidth<short>,
+          ISignedNumber<short>
+    {
+    }
+
+    public struct Int32
+        : IBinaryInteger<int>,
+          IConvertible,
+          IFixedWidth<int>,
+          ISignedNumber<int>
+    {
+    }
+
+    public struct Int64
+        : IBinaryInteger<long>,
+          IConvertible,
+          IFixedWidth<long>,
+          ISignedNumber<long>
+    {
+    }
+
+    public struct IntPtr
+        : IBinaryInteger<nint>,
+          IFixedWidth<nint>,
+          ISerializable,
+          ISignedNumber<nint>
+    {
+    }
+
+    public struct SByte
+        : IBinaryInteger<sbyte>,
+          IConvertible,
+          IFixedWidth<sbyte>,
+          ISignedNumber<sbyte>
+    {
+    }
+
+    public struct Single
+        : IBinaryFloatingPoint<float>,
+          IConvertible,
+          IFixedWidth<float>
+    {
+    }
+
+    public struct TimeOnly
+        : IComparableOperators<TimeOnly, TimeOnly>,
+          IFixedWidth<TimeOnly>,
+          ISpanFormattable,
+          ISpanParseable<TimeOnly>,
+          ISubtractable<TimeOnly, TimeOnly, TimeSpan>
+    {
+    }
+
+    public struct TimeSpan
+        : IAddable<TimeSpan, TimeSpan, TimeSpan>,
+          IComparableOperators<TimeSpan, TimeSpan>,
+          IDivisable<TimeSpan, double, TimeSpan>,
+          IDivisable<TimeSpan, TimeSpan, double>,
+          IFixedWidth<TimeSpan>,
+          IMultipliable<TimeSpan, double, TimeSpan>,
+          INegatable<TimeSpan, TResult>,
+          ISpanFormattable,
+          ISubtractable<TimeSpan, TimeSpan, TimeSpan>
+    {
+    }
+
+    public struct UInt16
+        : IBinaryInteger<ushort>,
+          IConvertible,
+          IFixedWidth<ushort>,
+          IUnsignedNumber<ushort>
+    {
+    }
+
+    public struct UInt32
+        : IBinaryInteger<uint>,
+          IConvertible,
+          IFixedWidth<uint>,
+          IUnsignedNumber<uint>
+    {
+    }
+
+    public struct UInt64
+        : IBinaryInteger<ulong>,
+          IConvertible,
+          IFixedWidth<ulong>,
+          IUnsignedNumber<ulong>
+    {
+    }
+
+    public struct UIntPtr
+        : IBinaryInteger<nuint>,
+          IFixedWidth<nuint>,
+          ISerializable,
+          IUnsignedNumber<nuint>
+    {
+    }
+}
+```
+
+### Pending Concepts
+
+There are several types which may benefit from some interface support. These include, but aren't limited to:
+* System.Array
+* System.DateOnly
+* System.Enum
+* System.Index
+* System.Range
+* System.TimeOnly
+* System.Tuple
+* System.ValueTuple
+* System.Numerics.BigInteger
+* System.Numerics.Complex
+* System.Numerics.Matrix3x2
+* System.Numerics.Matrix4x4
```

Comment:
There are lots of comments throughout the interfaces about how various decisions are being made in order to support types like matrices, so I'd hope we either implement the interfaces on matrices or don't constrain ourselves by things we aren't ourselves willing to do :)

Reply:
This list is more a `todo` than a "maybe we should think about these".

Even if we decided to not support our own types, for whatever reason, ensuring the interfaces work for 3rd party types is an important scenario.
(I do plan on defining interfaces and including them in the design doc here, but since I only got compiler bits a couple days ago, it was relatively slow going to "validate" the designs 😄)